src/main/webapp/app/exercises/shared/result/result-progress-bar/result-progress-bar.component.ts

```diff
@@ -0,0 +1,126 @@
+import { Component, OnDestroy, effect, input } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { ArtemisSharedCommonModule } from 'app/shared/shared-common.module';
+import { faCircleNotch } from '@fortawesome/free-solid-svg-icons';
+
+@Component({
+    selector: 'jhi-result-progress-bar',
+    standalone: true,
+    imports: [CommonModule, ArtemisSharedCommonModule],
+    templateUrl: './result-progress-bar.component.html',
+    styleUrl: './result-progress-bar.component.scss',
+})
+export class ResultProgressBarComponent implements OnDestroy {
+    estimatedRemaining = input<number>(0);
+    estimatedDuration = input<number>(0);
+    isBuilding = input<boolean>();
+    isQueued = input<boolean>();
```

Comment:
Are the inputs required? If yes make it explicit (does not need to be changed, just a general comment) 
```suggestion
    isBuilding = input.required<boolean>();
    isQueued = input.required<boolean>();
```